shared/contexts/WalletContext.tsx

```diff
@@ -40,20 +63,53 @@ export function WalletContextProvider (props: PropsWithChildren<any>): JSX.Eleme
   }, [provider, network, client])
 
   useEffect(() => {
-    wallet.get(0).getAddress().then(value => {
-      setAddress(value)
-    })
+    getWalletDetails()
```

Comment:
https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-wallet/src/wallet.ts#L77

You need to discover the `addresses` automatically when the user recovers their wallet and set the max/length.

Not here but somewhere in the recover wallet screens.